server/graphql/v1/mutations/orders.js

```diff
@@ -68,6 +80,30 @@ export async function createOrder(order, loaders, remoteUser) {
       );
     }
 
+    // Check if FromCollective reached max limit of orders per hour
+    if (ordersFromCollectiveCache.get(order.fromCollective.id)) {
+      if (ordersFromCollectiveCache.get(order.fromCollective.id) > 10) {
+        debug('Orders Cache: You have reached the hourly limit of donations', order);
+        throw new Error('You have reached the hourly limit of donations');
+      }
+      ordersFromCollectiveCache.set(order.fromCollective.id, ordersFromCollectiveCache.get(order.fromCollective.id) + 1);
+    } else {
+      ordersFromCollectiveCache.set(order.fromCollective.id, 1);
+    }
+    // Check if pair (FromCollective, Collective) reached max limit of orders per hour
+    if (ordersFromCollectiveAndCollectiveCache.get(`${order.fromCollective.id}_${order.collective.id}`)) {
+      if (ordersFromCollectiveAndCollectiveCache.get(`${order.fromCollective.id}_${order.collective.id}`) > 2) {
+        debug('Orders Cache: You have reached the hourly limit of donations for this Collective', order);
+        throw new Error('You have reached the hourly limit of donations for this Collective');
+      }
+      ordersFromCollectiveAndCollectiveCache.set(
```

Comment:
I recommend to put the cacheKey in a variable so that you can use it in the get and in the set without repeating it.